newa/__init__.py

```diff
@@ -311,6 +335,62 @@ class Recipe(Cloneable, Serializable):
     url: str
 
 
+@define
+class RecipeConfig(Cloneable, Serializable):
+    """ A job recipe configuration """
+
+    fixtures: dict = {}
+    dimensions: dict = {}
```

Comment:
* mutable defaults require `factory`, `field(factory=dict)`.
* types for keys and values are needed. Keys seem to be strings, values are what, lists of mappings? I think this deserves some `TypedDict` definitions, or even actual classes, but at least `TypedDict` to define "context" and "environment" bits so we could then re-use and compose in `dimensions` and other complex attributes. Right now, it's very open.

Reply:
I have added more types, making pre-commit happy. I am afraid that further improvements will require your personal involvement. :-)

Reply:
@happz Hi, would you find the current state OK-ish? :-)

Reply:
Checking... Although I will probably come with a commit, because I think we should be very careful with `Any` :)